CnFromPath must correctly handle the case when the path is absent

Instead of throwing an exception:
```
java.lang.RuntimeException: java.nio.file.NoSuchFileException: /some-unexisting-path
	at oo.atom.codegen.cn.CnFromPath.classNames(CnFromPath.java:49)
	at oo.atom.codegen.cn.AssertZeroClassNames.check(AssertZeroClassNames.java:15)
	at oo.atom.tests.TestCase.execute(TestCase.java:53)
	at oo.atom.tests.TestsSuite.lambda$null$0(TestsSuite.java:63)
	at org.junit.jupiter.engine.descriptor.JupiterTestDescriptor.executeAndMaskThrowable(JupiterTestDescriptor.java:141)
	at org.junit.jupiter.engine.descriptor.DynamicTestTestDescriptor.execute(DynamicTestTestDescriptor.java:41)
	at org.junit.jupiter.engine.descriptor.DynamicTestTestDescriptor.execute(DynamicTestTestDescriptor.java:24)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$3(HierarchicalTestExecutor.java:113)
	at org.junit.platform.engine.support.hierarchical.SingleTestExecutor.executeSafely(SingleTestExecutor.java:66)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.executeRecursively(HierarchicalTestExecutor.java:108)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.execute(HierarchicalTestExecutor.java:79)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$0(HierarchicalTestExecutor.java:115)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.junit.jupiter.engine.descriptor.TestFactoryTestDescriptor.lambda$invokeTestMethod$1(TestFactoryTestDescriptor.java:92)
	at org.junit.jupiter.engine.execution.ThrowableCollector.execute(ThrowableCollector.java:40)
	at org.junit.jupiter.engine.descriptor.TestFactoryTestDescriptor.invokeTestMethod(TestFactoryTestDescriptor.java:79)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:113)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:58)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$3(HierarchicalTestExecutor.java:113)
	at org.junit.platform.engine.support.hierarchical.SingleTestExecutor.executeSafely(SingleTestExecutor.java:66)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.executeRecursively(HierarchicalTestExecutor.java:108)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.execute(HierarchicalTestExecutor.java:79)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$2(HierarchicalTestExecutor.java:121)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$3(HierarchicalTestExecutor.java:121)
	at org.junit.platform.engine.support.hierarchical.SingleTestExecutor.executeSafely(SingleTestExecutor.java:66)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.executeRecursively(HierarchicalTestExecutor.java:108)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.execute(HierarchicalTestExecutor.java:79)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$2(HierarchicalTestExecutor.java:121)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$3(HierarchicalTestExecutor.java:121)
	at org.junit.platform.engine.support.hierarchical.SingleTestExecutor.executeSafely(SingleTestExecutor.java:66)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.executeRecursively(HierarchicalTestExecutor.java:108)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.execute(HierarchicalTestExecutor.java:79)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:55)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:43)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:170)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:154)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:90)
	at org.junit.platform.surefire.provider.JUnitPlatformProvider.invokeSingleClass(JUnitPlatformProvider.java:148)
	at org.junit.platform.surefire.provider.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:130)
	at org.junit.platform.surefire.provider.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:109)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
Caused by: java.nio.file.NoSuchFileException: /some-unexisting-path
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
	at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144)
	at sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
	at java.nio.file.Files.readAttributes(Files.java:1737)
	at java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:219)
	at java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276)
	at java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:322)
	at java.nio.file.FileTreeIterator.<init>(FileTreeIterator.java:72)
	at java.nio.file.Files.find(Files.java:3687)
	at oo.atom.codegen.cn.CnFromPath.classNames(CnFromPath.java:42)
	... 62 more
```
It must produce emply list of class names.

--- a/atom-basis/src/main/java/oo/atom/codegen/cn/AssertClassNamesContainCertainNames.java
+++ b/atom-basis/src/main/java/oo/atom/codegen/cn/AssertClassNamesContainCertainNames.java
@@ -1,0 +1,27 @@
+package oo.atom.codegen.cn;
+
+import io.vavr.collection.HashSet;
+import io.vavr.collection.List;
+import io.vavr.collection.Set;
+import oo.atom.tests.Assertion;
+import org.assertj.core.api.Assertions;
+
+public class AssertClassNamesContainCertainNames implements Assertion {
+    private final ClassNames classNames;
+    private final Set<String> namesToCheck;
+
+    public AssertClassNamesContainCertainNames(ClassNames classNames, Set<String> namesToCheck) {
+        this.classNames = classNames;
+        this.namesToCheck = namesToCheck;
+    }
+
+    public AssertClassNamesContainCertainNames(ClassNames classNames, String namesToCheck) {
+        this(classNames, HashSet.of(namesToCheck));
+    }
+
+    @Override
+    public final void check() throws Exception {
+        List<String> extractedNames = classNames.classNames();
+        Assertions.assertThat(namesToCheck).containsAll(extractedNames);
+    }
+}

--- a/atom-basis/src/main/java/oo/atom/codegen/cn/AssertZeroClassNames.java
+++ b/atom-basis/src/main/java/oo/atom/codegen/cn/AssertZeroClassNames.java
@@ -1,0 +1,17 @@
+package oo.atom.codegen.cn;
+
+import oo.atom.tests.Assertion;
+import org.assertj.core.api.Assertions;
+
+public class AssertZeroClassNames implements Assertion {
+    private final ClassNames classNames;
+
+    public AssertZeroClassNames(ClassNames classNames) {
+        this.classNames = classNames;
+    }
+
+    @Override
+    public final void check() throws Exception {
+        Assertions.assertThat(classNames.classNames()).isEmpty();
+    }
+}

--- a/atom-basis/src/main/java/oo/atom/codegen/cn/CnFromPath.java
+++ b/atom-basis/src/main/java/oo/atom/codegen/cn/CnFromPath.java
@@ -39,6 +39,9 @@ public class CnFromPath implements ClassNames {
     @Override
     public final List<String> classNames() {
         try {
+            if(Files.notExists(path)) {
+                return List.empty();
+            }
             final List<String> classes = Files.find(path, Integer.MAX_VALUE, (p, bf) -> p.toString().endsWith(".class"))
                 .map(path::relativize)
                 .map(Object::toString)

--- a/atom-basis/src/test/java/oo/atom/codegen/cn/CnFromPathTest.java
+++ b/atom-basis/src/test/java/oo/atom/codegen/cn/CnFromPathTest.java
@@ -1,0 +1,19 @@
+package oo.atom.codegen.cn;
+
+import oo.atom.tests.TestCase;
+import oo.atom.tests.TestsSuite;
+
+import java.nio.file.Paths;
+
+class CnFromPathTest extends TestsSuite {
+    public CnFromPathTest() {
+        super(
+            new TestCase(
+                "doesn't fail if the directory is absent",
+                new AssertZeroClassNames(
+                    new CnFromPath(Paths.get("/some-unexisting-path"))
+                )
+            )
+        );
+    }
+}

